<launch>

  <!-- Arguments -->
  <arg name="config_path" default="$(find okvis_ros)/../config/config_stereorig_v2.yaml" />

  <param name="/use_sim_time" value="true" />


  <!-- To un-compress image topics -->
  <node name="stereo_sync" type="stereo_sync" pkg="okvis_ros" output="screen">
    <param name="left_img_topic" value="/slave1/image_raw" />
    <param name="right_img_topic" value="/slave2/image_raw" />
    <param name="compressed" value="true" />
  </node>

  <!-- Run pose_graph node-->
  <node name="pose_graph_node" pkg="pose_graph" type="pose_graph_node">
    <param name="config_file" type="string" value="$(arg config_path)" />
  </node>


  <node name="okvis_node" pkg="okvis_ros" type="okvis_node">

    <param name="config_filename" value="$(arg config_path)" />
    <param name="mesh_file" value="firefly.dae" />


    <remap from="/camera0" to="/cam0/image_raw" />
    <remap from="/camera1" to="/cam1/image_raw" />

    <remap from="/imu" to="/imu/imu" />

  </node>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find okvis_ros)/config/rviz_svin.rviz" />

</launch>
